modules/radio/react/spec/Radio.spec.tsx

```diff
@@ -1,108 +1,104 @@
+  describe('when rendered without an id', () => {
+    it('should create a unique id for each instance', async () => {
+      const {getByLabelText} = render(
+        <form>
+          <Radio checked={true} onChange={cb} disabled={false} label="label1" />;
+          <Radio onChange={cb} disabled={false} label="label2" />;
+        </form>
+      );
 
-    expect(cb.mock.calls.length).toBe(1);
-    component.unmount();
-  });
+      const id1 = getByLabelText('label1').getAttribute('id');
+      const id2 = getByLabelText('label2').getAttribute('id');
 
-  test('Radio input should spread extra props', () => {
-    const component = mount(<Radio data-propspread="test" onChange={jest.fn()} />);
-    const input = component
-      .find('input') // TODO: Standardize on prop spread location (see #150)
-      .getDOMNode();
-    expect(input.getAttribute('data-propspread')).toBe('test');
-    component.unmount();
-  });
+      expect(id1).not.toEqual(id2);
+    });
+
+    it('should keep the same unique id if re-rendered', () => {
+      const {getByRole, rerender} = render(<Radio checked={false} onChange={cb} />);
 
-  test('Radio creates a unique id for each instance', async () => {
-    const fragment = mount(
-      <form>
-        <Radio checked={true} onChange={jest.fn()} disabled={false} />;
-        <Radio onChange={jest.fn()} disabled={false} />;
-      </form>
-    );
-
-    const id1 = fragment
-      .find('input')
-      .at(0)
-      .getDOMNode()
-      .getAttribute('id');
-
-    const id2 = fragment
-      .find('input')
-      .at(1)
-      .getDOMNode()
-      .getAttribute('id');
-
-    expect(id1).not.toEqual(id2);
-    fragment.unmount();
+      const uniqueId = getByRole('radio').getAttribute('id');
+      expect(getByRole('radio')).toHaveProperty('id', uniqueId);
+
+      rerender(<Radio checked={true} onChange={cb} />);
+
+      expect(getByRole('radio')).toHaveProperty('checked');
+      expect(getByRole('radio')).toHaveProperty('id', uniqueId);
+    });
   });
-});
 
-describe('Radio Accessibility', () => {
-  test('Radio should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <Radio id={'123'} label={'Label'} onChange={jest.fn()} />
-    );
-    expect(await axe(html)).toHaveNoViolations();
+  describe('when rendered with extra, arbitrary props', () => {
+    it('should spread extra props onto the radio', () => {
+      const attr = 'test';
+      const {getByRole} = render(<Radio data-propspread={attr} onChange={cb} />);
+      expect(getByRole('radio')).toHaveAttribute('data-propspread', attr);
+    });
   });
 
-  test('Radio without a defined id should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(<Radio label={'Label'} onChange={jest.fn()} />);
-    expect(await axe(html)).toHaveNoViolations();
+  describe('when rendered with an input ref', () => {
+    it('input ref should be defined', () => {
```

Comment:
@lychyi Found a bug with this logic and the input ref is always defined even if it's not pointing to the correct element. Can you check for non-null and look for `type="radio"`? (you can reference the checkbox tests)